Library/Formula/fish.rb

```diff
@@ -21,6 +21,9 @@ class Fish < Formula
 
   def install
     system "autoconf" if build.head?
+    inreplace "fish.pc.in" do |s|
+      s.gsub! /^completionsdir=.*$/, "completionsdir=#{HOMEBREW_PREFIX}/share/fish/vendor_completions.d"
```

Comment:
Fish [release 2.2.0](https://github.com/fish-shell/fish-shell/releases/tag/2.2.0) said:

> Fish now defines a default directory for other packages to provide completions. By default this is `/usr/share/fish/vendor-completions.d`; on systems with `pkgconfig` installed this path is discoverable with `pkg-config --variable completionsdir fish`.

Follow fish's guide, this patch customized fish completions dir by modify `fish.pc` to meet homebrew's directory requirment. In other words, let fish not use fish's default `/usr/share/fish/vendor_completions.d` but homebrew's `/usr/local/share/fish/vendor_completions.d`.